Fix cropping for wrapped images

maarzt opened the pull request:
Within the legacy user interface ij.plugin.Resizer is used to crop an image.
This currently produces wrap results.
This PR adds a functional implementation of AbstractVirtualStack#deleteSlice
to fix this issue. (Only the deletion of first and last slice is supported currently)

tpietzsch commented on src/main/java/net/imglib2/img/display/imagej/AbstractVirtualStack.java:
```diff
@@ -120,19 +141,24 @@ public void addSlice( final String sliceLabel, final ImageProcessor ip, final in
 
-	@Override
-	public void deleteLastSlice()
+	private void deleteFirtSlice()
```
Typo Firt --> First

tpietzsch commented on src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java:
```diff
@@ -150,14 +150,14 @@ public boolean isWritable()
 
 	@Override
-	public Object getPixels( final int n )
+	public Object getPixelsZeroBasedIndex( final int index )
```
I think this should be `protected`?

tpietzsch commented on src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java:
```diff
@@ -150,14 +150,14 @@ public boolean isWritable()
 
 	@Override
-	public void setPixels( final Object pixels, final int n )
+	public void setPixelsZeroBasedIndex( final int index, final Object pixels )
```
I think this should be protected?

tpietzsch commented on src/main/java/net/imglib2/img/display/imagej/PlanarImgToVirtualStack.java:
```diff
@@ -148,9 +152,33 @@ private static int initSize( final Interval interval )
+		img.setPlane( index, ( A ) arrayDataAccess );
+	}
+
+	private ArrayDataAccess wrapPixelsToAcces( Object pixels )
```
Typo. `Acces` --> `Access`

tpietzsch commented:
Please fix stuff that I commented.

This PR also adds implementation of `setPixels`. Could you comment on that? Is this also related to making `ij.plugin.Resizer` work?

maarzt commented:
@tpietzsch Thanks for reviewing this. Looks like I had a bad typing day. I will fix it.

maarzt commented:
> This PR also adds implementation of setPixels. Could you comment on that? Is this also related to making ij.plugin.Resizer work?

No these changes are not required to make ij.plugin.Resizer work.

In `AbstractVirtualStack` the `setPixels` method is now implemented, to make it easier for derived classes to correctly provide the `setPixels` functionality. It only has an effect, if the derived class implements `setPixelsZeroBasedIndex`.

The `PlanarImgToVirtualStack` class now implements exectly this `setPixelsZeroBasedIndex` method. This is a new feature. I split the PR into two commits such that this becomes more clear.

maarzt commented:
@tpietzsch I did all the changes you requested

tpietzsch commented:
👍
